scaleway/helpers_instance.go

```diff
@@ -159,3 +160,72 @@ func reachState(ctx context.Context, instanceAPI *instance.API, zone scw.Zone, s
 	}
 	return nil
 }
+
+// getServerType is a util to get a instance.ServerType by its commercialType
+func getServerType(apiInstance *instance.API, zone scw.Zone, commercialType string) *instance.ServerType {
+	serverType := (*instance.ServerType)(nil)
+
+	serverTypesRes, err := apiInstance.ListServersTypes(&instance.ListServersTypesRequest{
+		Zone: zone,
+	})
+	if err != nil {
+		l.Warningf("cannot get server types: %s", err)
+	} else {
+		serverType = serverTypesRes.Servers[commercialType]
+		if serverType == nil {
+			l.Warningf("unrecognized server type: %s", commercialType)
+		}
+	}
+
+	return serverType
+}
+
+// validateLocalVolumeSizes validates the total size of local volumes.
+func validateLocalVolumeSizes(volumes map[string]*instance.VolumeTemplate, serverType *instance.ServerType, commercialType string) error {
+	// Calculate local volume total size.
+	var localVolumeTotalSize scw.Size
+	for _, volume := range volumes {
+		if volume.VolumeType == instance.VolumeVolumeTypeLSSD {
+			localVolumeTotalSize += volume.Size
+		}
+	}
+
+	volumeConstraint := serverType.VolumesConstraint
+
+	// If no root volume provided, count the default root volume size added by the API.
+	if rootVolume := volumes["0"]; rootVolume == nil {
+		localVolumeTotalSize += volumeConstraint.MinSize
+	}
+
+	if localVolumeTotalSize < volumeConstraint.MinSize || localVolumeTotalSize > volumeConstraint.MaxSize {
+		min := humanize.Bytes(uint64(volumeConstraint.MinSize))
+		if volumeConstraint.MinSize == volumeConstraint.MaxSize {
+			return fmt.Errorf("%s total local volume size must be equal to %s", commercialType, min)
+		}
+
+		max := humanize.Bytes(uint64(volumeConstraint.MaxSize))
+		return fmt.Errorf("%s total local volume size must be between %s and %s", commercialType, min, max)
+	}
+
+	return nil
+}
+
+// sanitizeVolumeMap removes extra data for API validation.
+func sanitizeVolumeMap(serverName string, volumes map[string]*instance.VolumeTemplate) map[string]*instance.VolumeTemplate {
```

Comment:
Why do we need this ? 
Can we construct the struct correclty in the first place ?

Reply:
I added documentation to this function. I want to
1. Remain consistent with what is done in the CLI 
2. It allows us to manage multiple cases of entries and multiple validations in a single place. At least the advantage is that if their constraints change we have a specific place or apply the cleaning functions that depend on several different constraints (image specified or not, volume already existing or not).